apalis-core/src/worker/context.rs

```diff
@@ -376,3 +382,116 @@ impl Drop for WorkerContext {
         }
     }
 }
+
+#[cfg(test)]
+mod tests {
```

Comment:
TBH these tests look like overkill. They also dont actually test a worker.
I would recommend writing test(s) and using `run_with_ctx`.
```rs
pub async fn run_with_ctx(self, ctx: &mut WorkerContext) -> Result<(), WorkerError> {
```
You could run the worker separately and then control it with the context ref.